### ISSUE TITLE
Build on Mac OS fails. Could NOT find PTLS (missing: PTLS_FUSION_LIBRARY).

### ISSUE THREAD

**Issue by Mickdep** [private-octopus/picoquic#1302]
- Installed and built OpenSSL with brew.
- Installed PicoTLS by following the instructions in the README.

Both steps were completed without any errors.

The PicoTLS folder is placed on the same directory level as picoquic.

e.g.:

```
$ ls
picoquic/        picotls/
```

Building picoquic with the instructions in the README fails with the following output.

```
$ cmake .                                                                                                           
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Error at /opt/homebrew/Cellar/cmake/3.22.2/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find PTLS (missing: PTLS_FUSION_LIBRARY)
Call Stack (most recent call first):
  /opt/homebrew/Cellar/cmake/3.22.2/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindPTLS.cmake:19 (find_package_handle_standard_args)
  CMakeLists.txt:173 (find_package)


-- Configuring incomplete, errors occurred!
```

Is this a known issue? I was unable to find anything related to this.

**Comment by huitema** [comment]
First thing I hear about it. The builds are all tested on MAC OS for validation, see for example: https://github.com/private-octopus/picoquic/runs/5443550282?check_suite_focus=true.

This may be related to the switch by Apple from Intel CPU to M11 for some MAC -- the MAC OS tests on GitHub are run on 64 bit x86 VMs. The PicoTLS "Fusion" code was developed for the Intel architecture, and depends on the x86 instructions for accelerated AES. Are you observing this on an M11 MAC?

**Comment by Mickdep** [comment]
Ah that might explain things. I'm doing this on an M1 Mac. 
Knowing that's the case, I'll try to do things in an Ubuntu VM tomorrow.

**Comment by huitema** [comment]
We still need to debug this and find a solution. The ideal solution would be to add support for M1 in the picotls fusion code, but that may require some fancy coding. The second best solution would be to not have the dependency on PTLS_FUSION when compiling on M1. I would like a PR that does that, make PTLS_FUSION optional when compiling on M1, but I don't have access to M1, so someone else will have to do it!

**Comment by defermelowie** [comment]
I hit the same error when compiling for an ARM Cortex-A9 running Ubuntu. Is there a known work around or does picoquic simply not compile for arm at the moment?

**Comment by huitema** [comment]
@defermelowie, @Mickdep, I am working on a potential fix, see PR #1307 . I added a test to compile Picotls without the "fusion" option. I modified the CMaleLists.txt file to detect absence of the PTLS_FUSION library, and in that case compile Picoquic without reference to the fusion code. The test pass, but this is a test on Ubuntu with 64bit x86. I have not tested on M1, or on 32 bit ARM. Can you try downloading the PR branch and see whether it compiles on your systems?

**Comment by huitema** [comment]
@defermelowie Thanks! I am doing a slightly different fix, because PTLS_FUSION is in fact supported on some variants of __arm__:
```
#if (!defined(_WINDOWS) || defined(_WINDOWS64) && !defined(PTLS_WITHOUT_FUSION))
#include "picotls/fusion.h"
#endif
```
The flag `PTLS_WITHOUT_FUSION` is automatically set by CMake if the "ptls_fusion" library is not found in the Picotls build.

**Comment by huitema** [comment]
@defermelowie , If you can try again, that would validate the fix.

**Comment by defermelowie** [comment]
@huitema, I will check it tomorrow morning

**Comment by defermelowie** [comment]
@huitema it doesn't compile due to a mistake with the brackets I think, at [picoquic/tls_api.c:52](https://github.com/private-octopus/picoquic/blob/ee13f08adbc65bc184e6b5050f4ec32af5939a3d/picoquic/tls_api.c#L52). The current version is:
```C
#if (!defined(_WINDOWS) || defined(_WINDOWS64) && !defined(PTLS_WITHOUT_FUSION))
#include "picotls/fusion.h"
#endif
```
I got it comping by changing it to:
```C
#if (!defined(_WINDOWS) || defined(_WINDOWS64)) && !defined(PTLS_WITHOUT_FUSION)
#include "picotls/fusion.h"
#endif
```

**Comment by huitema** [comment]
Damn, I thought I fixed that! It does compile on Ubuntu, go figure...

Thanks for testing!

**Comment by huitema** [comment]
Fixed in main branch with PR #1307
